### DISCUSSION TITLE
HTTP 400 using Send REST API

### DISCUSSION THREAD

**Discussion by sipatha** [postalserver/postal#2373] [Help with using Postal]
Hi,
I have an interesting situation where if the Send API is called from the server I get HTTP 400 response and no emails are sent. However if I take the same Send message and same API Key, I am able to send using my local Postman.

I have gone as far as to create a new API Key and use it exclusively for the server. Again I received an HTTP 400 error, and usage of the API Key was usage was updated. The usual test using Postman also worked on my local machine using the new API Key

``` json
{
    "to": ["automation@thedomain.com"],
    "bcc": ["dad@thedomain.com","mum@thedomain.com"],
    "sender": "home-notifications@email.thedomain.com",
    "from": "home-notifications@email.thedomain.com",
    "subject": "Zone X Sprinklers Triggered",
    "plain_body": "Zone X Sprinklers Triggered, soil moisture level at X"
}
```

**Comment by willpower232** [comment]
What is the content of the HTTP 400 response? Are you sending the message from something else on the same server as Postal?

**Reply by sipatha** [reply]
The response body is empty.

I've noted that the server always returns HTTP 200 for both success and error responses.

BTWN: I'm on version 2.1.1

**Reply by willpower232** [reply]
if the response body is empty, are you definitely reaching Postal and not anything else you happen to be hosting? I don't think there is an empty response body 400

**Comment by ghost** [comment]
FYI, I had a similar issue but the cause was formatting, I had a trailing comma in my json post data, ie:

curl -v -X POST https://HOST/api/v1/send/message \
     -H "Content-Type: application/json" \
     -H "X-Server-API-Key: REDACTED" \
     -d '{
           "to": ["user@domain.com"],
           "sender": "user@otherdomain.com",
           "from": "user@otherdomain.com",
           "subject": "Test",
           "plain_body": "Test",
         }'

Threw the null 400, but:

curl -v -X POST https://HOST/api/v1/send/message \
     -H "Content-Type: application/json" \
     -H "X-Server-API-Key: REDACTED" \
     -d '{
           "to": ["user@domain.com"],
           "sender": "user@otherdomain.com",
           "from": "user@otherdomain.com",
           "subject": "Test",
           "plain_body": "Test"
         }'

Did not. The issue was the trailing comma on plain_body.
